### PR TITLE
Bettering torrent support.

### LINKED ISSUE #1609: Fix torrent magnet links, add a helper link for magnet URLs.
### Pre-Flight checklist

- [X] Did you check to see if this issue already exists?
- [X] This is a single feature request. (Do not put multiple feature requests in one issue)
- [X] This is not a question or discussion. (Use https://lemmy.ml/c/jerboa for that)

### Describe The Feature Request Below

I checked again recently, and magnet links don't seem to be working correctly atm. 

I've added a helper link to lemmy's docs [here](https://join-lemmy.org/docs/users/02-media.html#torrents), that I'll try to figure out how to link to.

### PR DESCRIPTION
**Pull request by dessalines**
- Fixes URI validation to include magnet link schemes.
- When creating a post, can automatically copy the torrent title from the link.
- Adds a torrent helper for magnet links.
- Fixes magnet link clicks.
- Fixes #1609

Pics:

Screenshot_2024-08-05-16-10-51-73_c9298fda48802cf3e2d17cb10a48759f

Screenshot_2024-08-05-16-11-05-91_c9298fda48802cf3e2d17cb10a48759f

### CODE REVIEW

File: app/src/main/java/com/jerboa/ui/components/post/PostListing.kt [hunk 1]
```diff
@@ -1789,3 +1794,24 @@ fun MetadataCard(post: Post) {
...
     }
 }
+
+@Composable
+fun TorrentHelpInfo() {
+    OutlinedCard(
+        shape = MaterialTheme.shapes.medium,
+        modifier =
+            Modifier
+                .padding(vertical = MEDIUM_PADDING, horizontal = MEDIUM_PADDING)
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/ui/components/post/PostListing.kt:1804]: Isn't this the same as `padding(MEDIUM_PADDING)`

File: app/src/main/java/com/jerboa/ui/components/post/create/CreatePostScreen.kt [hunk 2]
```diff
@@ -118,12 +119,24 @@ fun CreatePostScreen(
...
 
     val (suggestedTitle, suggestedTitleLoading) =
-        when (val res = createPostViewModel.siteMetadataRes) {
-            ApiState.Empty -> MetaDataRes(null, false)
-            ApiState.Loading -> MetaDataRes(null, true)
-            is ApiState.Success ->
-                MetaDataRes(res.data.metadata.title, false)
-            else -> MetaDataRes(null, false)
+        // If its a torrent magnet link, use the dn param to extract the torrent title
+        if (uri.scheme == "magnet") {
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/ui/components/post/create/CreatePostScreen.kt:123]: Can you add a constant for these strings.

**Review by MV-GH — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/jerboa/ui/components/post/create/CreatePostScreen.kt (hunk 2 above)
Comment (dessalines) [reply app/src/main/java/com/jerboa/ui/components/post/create/CreatePostScreen.kt:123]: I did add a `ALLOWED_SCHEMES` list, but this one only needs to check one of them.

File: app/src/main/java/com/jerboa/ui/components/post/PostListing.kt (hunk 1 above)
Comment (dessalines) [reply app/src/main/java/com/jerboa/ui/components/post/PostListing.kt:1804]: Good call.

**Comment by MV-GH** [comment]
You need to fix the tests too

**Review by MV-GH — APPROVED** [review]

**Pull request merged**